Spring-2019/arrays_sorting/product_of_array/GetProduct.java

```diff
@@ -0,0 +1,56 @@
+// Class Definition
+public class GetProduct {
+
+    // Solution Method Definition
+    // Note: this is the optimized solution
+    public int[] arrayProduct(int[] start) {
+        int total = 1;
+        int nonZeroTotal = 1;
+        int zeroCount = 0;
+        // get total product of array
+        for (int i = 0; i < start.length; i++) {
+            if (start[i] == 0) {
+                zeroCount++;
+                total = 0;
+            } else {
+                total *= start[i];
+                nonZeroTotal *= start[i];
+            }
+        }
+        // create new array to store result
+        int[] result = new int[start.length];
+        for (int i = 0; i < start.length; i++) {
+            if (start[i] == 0) {
+                if (zeroCount == 1) {
+                    result[i] = nonZeroTotal;
+                } else {
+                    result[i] = 0;
+                }
+            } else {
+                result[i] = total / start[i];
+            }
+        }
+        return result;
+    }
+
+    public String printArray(int[] array) {
+        String result = "";
+        for (int i = 0; i < array.length; i++) {
+            result += array[i];
+            result += " ";
```

Comment:
```java
            result += array[i] + " ";
```

Reply:
addressed in 71bf165f8b0986f60baca429d5ee724682357106
TY~